Cabal/Distribution/Types/PackageDescription.hs

```diff
@@ -369,22 +369,15 @@ withForeignLib pkg_descr f =
 -- dependencies.
 allBuildInfo :: PackageDescription -> [BuildInfo]
 allBuildInfo pkg_descr = [ bi | lib <- allLibraries pkg_descr
```

Comment:
There should be a mention in the changelog that the semantics of this function changed. Also the Haddock comment should be updated.